### ISSUE TITLE
Missing license

### ISSUE THREAD

**Issue by asperan** [mag37/dockcheck#20]
Hi!

I have seen your project on reddit and, while I can find it useful, it doesn't seem to have a license.

Because of this, this project is not usable/forkable/distributable/modifiable by any person other than the creator, you:

> To use a piece of software, whether it’s open source or commercial, you need some grant of rights. In the U.S. and many other places, creative work (including software) is protected by exclusive copyright by default. This means that no one can legally use, copy, distribute, or modify that software without explicit permission from the creator/author. This permission comes in the form of a license that grants the right to do so. Without that license, the baseline assumption is that you do not have permission to use the software.

([Source](https://www.synopsys.com/blogs/software-security/unlicensed-open-source-scenarios/))

I suggest you to [choose a license](https://choosealicense.com/licenses/), so that anybody can help you!

Anyway, good job and keep up the good work!

**Comment by mag37** [comment]
Oh you're right and I've totally forgot to read up on that - it's on my todo-list!

I've never released anything before and I'm a bit confused about the different licenses - and don't know what's seen as good/bad in the community.

Anywho - I guess I also need base it on what my script uses underneath.
`regctl` which is Apache 2.0 for example.

I need to read up on this before I decide between the usual GPL/Apache/MIT.


Thank you for the heads up. 
I gladly receive advice!

**Comment by mag37** [comment]
I'm leaning towards GPL due to strong copyleft and OSS mindset.

**Comment by asperan** [comment]
> I'm a bit confused about the different licenses

Don't worry, even if I seem firm on licenses, I still have doubts on every project I create :D

> what's seen as good/bad in the community

Someone could say GPL is too strict, others can say MIT allows "bad companies" to steal open source software. IMHO there is not a single truth, for example some of my projects are GPL licensed, other are MPL licensed. It really depends on what license fits your needs and what license you like.

>  I guess I also need base it on what my script uses underneath.
> regctl which is Apache 2.0 for example.

Yes, good catch! Your license of choice must be compatible with the license of (all) the third-party components you use.

> I'm leaning towards GPL due to strong copyleft and OSS mindset.

That's good!

**Comment by mag37** [comment]
Thank you for the insight.
Well - it seems like GPL is what I'd like and from my understanding Apache 2 as a (not baked in) dependency that the user downloads/get asked to download, should be fine unde GPL.   
Apache:
>Licensed works, modifications, and larger works may be distributed under different terms and without source code.

Thou how would I practically include the 3d party Apache License? I understand I'd make a License-file with the GPL, is it enough to mention in the docs/readme the dependency license reference with name+url?

Sorry for being totally green and thank you kindly for giving your input.

**Comment by asperan** [comment]
Well, what keeps me from having a sure answer is that your work is based on regctl, but it is not included in your repository: you download it in the user PC, so (IMO) it is not the same as a library of some sort.

> Thou how would I practically include the 3d party Apache License? I understand I'd make a License-file with the GPL, is it enough to mention in the docs/readme the dependency license reference with name+url?

Yes, you have to create a license file for the GPL license and, I'm not a lawyer, but I'd say that you can just list the dependencies (in your case, regctl) and say something along the lines of "This are the dependency and they are distributed/licensed according to their terms" (if their terms are compatible with the redistribution). I guess a link to the repository can be good.

In this way, your original work is licensed under GPL, and the redistributed dependencies use their original license.

> Sorry for being totally green

Don't be sorry :) It's ok to not know something and, as you can see, neither I know everything (I just know a little)!

**Comment by mag37** [comment]
Thank you for the reply.
You're confirming my conclusion, so I've added the GPL in a LICENSE file, github had some simple tooling templates for it.

Then I added a note about the dependency being Apache2 in the readme:

[`regclient/regctl`](https://github.com/regclient/regclient) (Licenced under [Apache-2.0 License](http://www.apache.org/licenses/LICENSE-2.0))   
User will be prompted to download `regctl` if not in `PATH` or `PWD`

---
Like that. And from what I could read that should suffice.

It's all in a new [branch](https://github.com/mag37/dockcheck/tree/v0.1.8) with a bunch of other changes which will be merged later today.

Thanks again for your time and input!
